crates/autopilot/src/run_loop.rs

```diff
@@ -89,7 +90,43 @@ impl RunLoop {
 
     async fn single_run_(&self, id: AuctionId, auction: &Auction) {
         tracing::info!("solving");
-        let mut solutions = self.solve(auction, id).await;
+        let solutions = self.solve(auction, id).await;
+
+        // Validate solutions and filter out invalid ones.
+        let mut solutions = solutions
+            .into_iter()
+            .filter_map(|(index, response)| {
```

Comment:
nit: can use `flat_map` here.

Reply:
How would I filter them then?

Reply:
Option is an iterator, so you can likely flat map as Felix suggested.